nautobot/core/templates/inc/nav_menu.html

```diff
@@ -111,3 +112,73 @@
         </div>
     </div>
 </nav>
+
+<script>
+    document.addEventListener('DOMContentLoaded', function() {
+        var navbar = document.querySelector('.navbar-fixed-left');
+        var dropdownToggles = document.querySelectorAll('.navbar-fixed-left .navbar-nav > .dropdown > a[data-toggle="collapse"]');
+        var dropdowns = document.querySelectorAll('.navbar-fixed-left .navbar-nav .collapse');
+        var lastDropdownId = localStorage.getItem('lastOpenedDropdown');
+        var savedScrollPosition = localStorage.getItem('navbarScrollPosition');
+
+        // Debounce function
+        function debounce(func, wait, immediate) {
+            var timeout;
+            return function() {
+                var context = this, args = arguments;
+                var later = function() {
+                    timeout = null;
+                    if (!immediate) func.apply(context, args);
+                };
+                var callNow = immediate && !timeout;
+                clearTimeout(timeout);
+                timeout = setTimeout(later, wait);
+                if (callNow) func.apply(context, args);
+            };
+        }
+
+        // Close all dropdowns except the one specified
+        function closeAllDropdownsExcept(exceptId) {
+            dropdowns.forEach(function(collapse) {
+                if (collapse.id !== exceptId && collapse.classList.contains('in')) {
+                    $(collapse).collapse('hide');
+                }
+            });
+        }
+
+        // Close all dropdowns except the last opened one
+        dropdownToggles.forEach(function(toggle) {
+            toggle.addEventListener('click', function(event) {
+                event.preventDefault();
+                var collapseElement = document.getElementById(this.getAttribute('href').substring(1));
+
+                if (!collapseElement.classList.contains('in')) {
+                    closeAllDropdownsExcept(collapseElement.id);
+                    $(collapseElement).collapse('show');
+                    localStorage.setItem('lastOpenedDropdown', collapseElement.id);
+                } else {
+                    $(collapseElement).collapse('hide');
+                    localStorage.removeItem('lastOpenedDropdown');
+                }
+            });
+        });
+
+        // Open the last opened dropdown
+        if (lastDropdownId) {
```

Comment:
I believe the purpose of this logic is to open the dropdown menu on the current page. 

However, I see some potential downsides. For example, if we open the `DEVICES` dropdown, and click on `Devices` to navigate to the devices list view, and then click the logo to return home, the `DEVICES` dropdown will remain open. Furthermore, if we manually change the URL in the address bar to navigate to a different page, the `DEVICES` dropdown remains.

Reply:
If I'm right about the intent, we could switch to a URL-based approach for determining the opened dropdown, rather than relying on the last dropdown clicked.

Reply:
Is it a bad thing if the menu stays open when you switch to a different page though?

Reply:
If it has nothing to do with the current page, I don't see why it should be open, and I haven't observed this pattern on any other dashboard I've visited.

I think that gives the user the wrong impression; it kind of appears like that's the current active dropdown.

Reply:
The intent was _not_ to match the menu to the currently opened page. 

It is a common UX design pattern for a "Side Navbar" to stay open on a current menu context.  For pages that are grouped together it is common to switch between topic areas.  I agree it would be uncommon for a Top mounted navigation to keep the menus collapsed, but sidebar navigation is a different experience and context is important to maintain.

Some easy to imagine examples would be switching between any of these items below:
* Locations -> Location Types
* Circuits -> Circuit Types
* Racks -> Rack Groups
* Power Feeds -> Power Panels
* IP Addresses -> Prefixes -> Namespaces -> etc...
* --- Many more examples here ---

Keeping the current context of the menu state and position of the scrollbar is more natural feeling to the user than click a link and having everything close back up only to have to find where you last were and then navigate all over from scratch.

One case I agree with modifying is when clicking the Nautobot icon (upper left).  When switching back to the homepage / panel view it would be smart to reset all menus to collapsed.

Reply:
Thinking of the longer-term UI direction I do think there's an argument for keeping the state of the sidebar nav independent from the state of the page contents.

Reply:
I agree with Brad, I have recently been working with Prefixes and IPAddresses and have found it annoying that the side navbar collapses once I click into one of those Models, and that to switching back and forth requires me to reopen the dropdown